pennylane/math/quantum.py

```diff
@@ -213,28 +214,32 @@ def _density_matrix_from_matrix(density_matrix, indices, check_state=None, c_dty
         if (
             len(density_matrix.shape) != 2
             or density_matrix.shape[0] != density_matrix.shape[1]
-            or np.ceil(np.log2(shape)) != np.floor(np.log2(shape))
+            or not np.log2(shape).is_integer()
         ):
             raise ValueError("Density matrix must be of shape (2**N, 2**N).")
         # Check trace
         trace = np.trace(density_matrix)
         if not is_abstract(trace):
             if not allclose(trace, 1.0, atol=1e-10):
                 raise ValueError("The trace of the density matrix should be one.")
+            # Check if the matrix is hermitian

```

Comment:
Do we also want to check the positive semi-definite property?

Reply:
I agree. An input like `np.array([[2, 0], [0, -1]])` is still allowed currently.

Reply:
Yes I agree I will add a check here 👍